### ISSUE TITLE
Ride stops working when running tests

### ISSUE THREAD

**Issue by yanne** [robotframework/RIDE#1184]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145) by martijn.... on 3 Dec 2012*

I was running ride as follows:
ride.py . &amp;
My test cases, located in the current directory, took 30 - 30 seconds to load.
I selected some test cases and pressed the run button. The tests ran fine until
a test suite was finished (it may be after the last selected test). The last
test was finished. Then the screen went gray and robot framework did not 
finish running the tests. 

After this test, the graphics of my virtual box was messed up. For example,
when I typed something in a command prompt, I did not see the text I was typing.
It appeared only after pressing enter. Graphics became normal after I pressed
the full-screen button of my virtual box (top-right on the virtual box window).
First I made it not full screen, then back to full screen.

My test cases start another application (called Terra) and connect to it
using TCP/IP. Terra has been written in C++. We use our own python code to 
connect to Terra.

Here are some versions:

Operating system: Ubuntu 10.04 running under virtual box.
Python version: 2.6.5.
Ride version 0.53.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c1) by @mkorpela on 3 Dec 2012*


Does this happen only when you execute the test cases through RIDE?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c2) by martijn.... on 4 Dec 2012*


Yes, this only happens when I run the tests with ride. The number of tests that run successfully before the screen becomes gray varies. I tried multiple tims.

While the ride screen was gray, I also tried ps -ef | grep "Terra" on a command prompt. Terra was not running anymore. Terra is started during suite start-up and suite tear-down.

After a faild attempt with ride, I observed tha the ride gui starts pybot with an argument file. I copied the argument file and removed the gui-related lnes from it. When I ran pybot with that argument file, the tests ran fine. Before that, I also tried to run the tests with pybot, providing other arguments that point to the same tests. That test was also successful.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c3) by @mkorpela on 6 Dec 2012*


How much memory do you have in that virtual machine ?

How much free memory do you have when executing the test cases?

What Robot Framework version are you using ( pybot --version ) ?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c4) by martijn.... on 7 Dec 2012*


Total memory of computer: 4096 MB
Memory in virtual box: 1973 MB

Memory statistics after loading the tests:

terra@&#8288;terra-vm:~$ cat /proc/meminfo
MemTotal:        1985728 kB
MemFree:         1038576 kB
Buffers:           65220 kB
Cached:           443744 kB
SwapCached:            0 kB
Active:           552124 kB
Inactive:         352080 kB
Active(anon):     395532 kB
Inactive(anon):    11240 kB
Active(file):     156592 kB
Inactive(file):   340840 kB
Unevictable:           0 kB
Mlocked:               0 kB
HighTotal:       1110984 kB
HighFree:         262136 kB
LowTotal:         874744 kB
LowFree:          776440 kB
SwapTotal:       1370104 kB
SwapFree:        1370104 kB
Dirty:               256 kB
Writeback:             0 kB
AnonPages:        395360 kB
Mapped:            92764 kB
Shmem:             11536 kB
Slab:              23768 kB
SReclaimable:      14424 kB
SUnreclaim:         9344 kB
KernelStack:        2376 kB
PageTables:         5368 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     2362968 kB
Committed_AS:    1024944 kB
VmallocTotal:     122880 kB
VmallocUsed:       31484 kB
VmallocChunk:      86412 kB
HardwareCorrupted:     0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       4096 kB
DirectMap4k:       12280 kB
DirectMap4M:      897024 kB
terra@&#8288;terra-vm:~$ 

Memory statistics while running:

terra@&#8288;terra-vm:~$ cat /proc/meminfo
MemTotal:        1985728 kB
MemFree:          981120 kB
Buffers:           65348 kB
Cached:           445388 kB
SwapCached:            0 kB
Active:           609016 kB
Inactive:         353036 kB
Active(anon):     451732 kB
Inactive(anon):    11872 kB
Active(file):     157284 kB
Inactive(file):   341164 kB
Unevictable:           0 kB
Mlocked:               0 kB
HighTotal:       1110984 kB
HighFree:         205096 kB
LowTotal:         874744 kB
LowFree:          776024 kB
SwapTotal:       1370104 kB
SwapFree:        1370104 kB
Dirty:               132 kB
Writeback:             0 kB
AnonPages:        451452 kB
Mapped:            93556 kB
Shmem:             12168 kB
Slab:              24060 kB
SReclaimable:      14712 kB
SUnreclaim:         9348 kB
KernelStack:        2416 kB
PageTables:         5516 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     2362968 kB
Committed_AS:    1096248 kB
VmallocTotal:     122880 kB
VmallocUsed:       31484 kB
VmallocChunk:      86412 kB
HardwareCorrupted:     0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       4096 kB
DirectMap4k:       12280 kB
DirectMap4M:      897024 kB
terra@&#8288;terra-vm:~$ 

pybot --version
Robot Framework 2.5.7 (Python 2.6.5 on linux2)

Memory statistics while running tests - later:

terra@&#8288;terra-vm:~$ cat /proc/meminfo
MemTotal:        1985728 kB
MemFree:          937840 kB
Buffers:           65996 kB
Cached:           477888 kB
SwapCached:            0 kB
Active:           617344 kB
Inactive:         384676 kB
Active(anon):     458428 kB
Inactive(anon):    11872 kB
Active(file):     158916 kB
Inactive(file):   372804 kB
Unevictable:           0 kB
Mlocked:               0 kB
HighTotal:       1110984 kB
HighFree:         163060 kB
LowTotal:         874744 kB
LowFree:          774780 kB
SwapTotal:       1370104 kB
SwapFree:        1370104 kB
Dirty:                96 kB
Writeback:             0 kB
AnonPages:        458132 kB
Mapped:           109440 kB
Shmem:             12168 kB
Slab:              24516 kB
SReclaimable:      15092 kB
SUnreclaim:         9424 kB
KernelStack:        2456 kB
PageTables:         5632 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     2362968 kB
Committed_AS:    1132620 kB
VmallocTotal:     122880 kB
VmallocUsed:       33324 kB
VmallocChunk:      52776 kB
HardwareCorrupted:     0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       4096 kB
DirectMap4k:       12280 kB
DirectMap4M:      897024 kB
terra@&#8288;terra-vm:~$ 

Memory statistics while ride is gray

terra@&#8288;terra-vm:~$ cat /proc/meminfo
MemTotal:        1985728 kB
MemFree:          947696 kB
Buffers:           66364 kB
Cached:           479676 kB
SwapCached:            0 kB
Active:           607356 kB
Inactive:         386868 kB
Active(anon):     448488 kB
Inactive(anon):    11872 kB
Active(file):     158868 kB
Inactive(file):   374996 kB
Unevictable:           0 kB
Mlocked:               0 kB
HighTotal:       1110984 kB
HighFree:         173228 kB
LowTotal:         874744 kB
LowFree:          774468 kB
SwapTotal:       1370104 kB
SwapFree:        1370104 kB
Dirty:               108 kB
Writeback:             4 kB
AnonPages:        448336 kB
Mapped:           109452 kB
Shmem:             12168 kB
Slab:              24560 kB
SReclaimable:      15132 kB
SUnreclaim:         9428 kB
KernelStack:        2456 kB
PageTables:         5596 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     2362968 kB
Committed_AS:    1129480 kB
VmallocTotal:     122880 kB
VmallocUsed:       31484 kB
VmallocChunk:      86412 kB
HardwareCorrupted:     0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       4096 kB
DirectMap4k:       12280 kB
DirectMap4M:      897024 kB
terra@&#8288;terra-vm:~$

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c5) by @mkorpela on 10 Dec 2012*


Could you update Robot Framework to 2.7.5? 2.5.x versions had a very bad performance during report and log generation. These might cause the problems you have described.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c6) by martijn.... on 13 Dec 2012*


Dear Mikko,

I tried to update robot framework but failed. I downloaded the file
robotframework-2.7.5.tar.gz
I unpacked this file and went into the directory robotframework-2.7.5. There I did
sudo python setup.py install

Checking my installation failed. I did pybot --version and the result was:

terra@&#8288;terra-vm:~/work/robotframework$ pybot --version
Traceback (most recent call last):
  File "/usr/local/bin/pybot", line 4, in &lt;module&gt;
    from robot import run_cli
ImportError: cannot import name run_cli
terra@&#8288;terra-vm:~/work/robotframework$ 

After this error, I extended my path and retried. My path is now:
/usr/local/bin/pybot:/usr/bin/python:/home/terra/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/jython2.5.2

I also investivated sys.path under python:
terra@&#8288;terra-vm:~$ python
Python 2.6.5 (<a href="https://code.google.com/p/robotframework-ride/source/detail?r=265">r265</a>:79063, Apr 16 2010, 13:09:56) 
[GCC 4.4.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
&gt;&gt;&gt; import sys
&gt;&gt;&gt; print sys.path
['', '/usr/local/lib/python2.6/dist-packages/Cheetah-2.4.4-py2.6.egg', '/usr/local/lib/python2.6/dist-packages/Markdown-2.1.1-py2.6.egg', '/usr/local/lib/python2.6/dist-packages/xlrd-0.7.7-py2.6.egg', '/usr/local/lib/python2.6/dist-packages/argparse-1.2.1-py2.6.egg', '/usr/local/lib/python2.6/dist-packages/protobuf-2.3.0-py2.6.egg', '/usr/local/lib/python2.6/dist-packages/robotframework-2.5.7-py2.6.egg', '/usr/local/lib/python2.6/dist-packages/robotframework_ride-0.53-py2.6.egg', '/usr/lib/python2.6', '/usr/lib/python2.6/plat-linux2', '/usr/lib/python2.6/lib-tk', '/usr/lib/python2.6/lib-old', '/usr/lib/python2.6/lib-dynload', '/usr/lib/python2.6/dist-packages', '/usr/lib/python2.6/dist-packages/PIL', '/usr/lib/python2.6/dist-packages/gst-0.10', '/usr/lib/pymodules/python2.6', '/usr/lib/python2.6/dist-packages/gtk-2.0', '/usr/lib/pymodules/python2.6/gtk-2.0', '/usr/lib/python2.6/dist-packages/wx-2.8-gtk2-unicode', '/usr/local/lib/python2.6/dist-packages']
&gt;&gt;&gt;

Furthermore I checked robot directory under my python installation directory:
terra@&#8288;terra-vm:/usr/local/lib/python2.6/dist-packages/robot$ ls
api     errors.py   __init__.py   jarrunner.pyc          libdocpkg   libraries  parsing               rebot.py   result      running  testdoc.py   tidy.pyc   version.py
common  errors.pyc  __init__.pyc  jythonworkarounds.py   libdoc.py   model      pythonpathsetter.py   rebot.pyc  runner.py   run.py   testdoc.pyc  utils      version.pyc
conf    htmldata    jarrunner.py  jythonworkarounds.pyc  libdoc.pyc  output     pythonpathsetter.pyc  reporting  runner.pyc  run.pyc  tidy.py      variables  writer
terra@&#8288;terra-vm:/usr/local/lib/python2.6/dist-packages/robot$

Do you know what I did wrong? 

With kind regards,

Martijn Dirkse

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c7) by @pekkaklarck on 13 Dec 2012*


Sounds like `pybot` would find your old Robot installation and not the new one. Could you try running the following command to see what `robot` module gets imported:

python -c "import robot; print robot.__file__; print robot.__version__"

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c8) by martijn.... on 14 Dec 2012*


terra@&#8288;terra-vm:~$ python -c "import robot; print robot.__file__; print robot.__version__"
/usr/local/lib/python2.6/dist-packages/robotframework-2.5.7-py2.6.egg/robot/__init__.pyc
2.5.7
terra@&#8288;terra-vm:~$ 

I think you are right.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c9) by @pekkaklarck on 14 Dec 2012*


You have most likely installed earlier version with easy_install which very annoyingly places that installation into the beginning of the sys.path when you run Python. If you have `pip` available, you could try `pip uninstall robotframework` as many times as it succeeds and `pip install robotframework` then.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c10) by @mkorpela on 14 Dec 2012*


If you don't have pip installed you can install it with easy_install:
easy_install pip

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c11) by martijn.... on 17 Dec 2012*


Dear Mikko and Pekka,

Thank you. I installed pip. Then I uninstalled robot framework using pip as you described. I reinstalled it and now I have the right version:

terra@&#8288;terra-vm:~/work/Platform1/ConnectedTerra/test/testRideInstallation$ pybot --version
Robot Framework 2.7.5 (Python 2.6.5 on linux2)
terra@&#8288;terra-vm:~/work/Platform1/ConnectedTerra/test/testRideInstallation$ 

I uninstalled and reinstalled robotframework-ride using pip. I created a very simpl test that does not use Python libraries of our own. I tried to run it from th ride gui. An error dialog was shown that the tests could not be run (or not be started, I forgot the exact wording). The ride display showed:

command:  --argumentfile /tmp/RIDEYnd23R.d/argfile.txt --listener /usr/local/lib/python2.6/dist-packages/robotframework_ride-0.53-py2.6.egg/robotide/contrib/testrunner/TestRunnerAgent.py:53998 /home/terra/work/Platform1/ConnectedTerra/test/testRideInstallation
[Errno 2] No such file or directory: ''

Here is some more information:

terra@&#8288;terra-vm:~/work/Platform1/ConnectedTerra/test/testRideInstallation$ gedit /tmp/RIDEYnd23R.d/argfile.txt &amp;

--outputdir
/tmp/RIDEYnd23R.d
--monitorcolors
off
--monitorwidth
219
--test
testRideInstallation.Suite.Test

terra@&#8288;terra-vm:~/work/Platform1/ConnectedTerra/test/testRideInstallation$ ls /tmp/RIDEYnd23R.d
argfile.txt

terra@&#8288;terra-vm:~/work/Platform1/ConnectedTerra/test/testRideInstallation$ /usr/local/lib/python2.6/dist-packages/robotframework_ride-0.53-py2.6.egg/robotide/contrib/testrunner/TestRunnerAgent.py &amp;
File exists and contains Python code

terra@&#8288;terra-vm:~/work/Platform1/ConnectedTerra/test/testRideInstallation$ ls /home/terra/work/Platform1/ConnectedTerra/test/testRideInstallation
os  socket  SocketServer  Suite.txt  threading
terra@&#8288;terra-vm:~/work/Platform1/ConnectedTerra/test/testRideInstallation$ 

Suite.txt contains the following data:

*** Test Cases ***
Test
    ${var1}=    Set Variable    aaa
    ${var2}=    Set Variable    bbb
    Should Be Equal    ${var1}${var2}    aaabbb

This problem is certainly different from what I experienced before upgrading, so I see progress. Can you help me with this issue?

With kind regards,

Martijn Dirkse

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c12) by @pekkaklarck on 17 Dec 2012*


Do you mean that running the simple test case in your previous comment fails after fresh Robot and RIDE installations? That is pretty strange and I really don't know why that would happen.

Only strange thing in the outputs you provided is how argfile.txt looks like. Does it really contain newlines after options i.e. are '--outputdir' and '/tmp/RIDEYnd23R.d' really on separate lines?

Another thing I noticed was that you executed only one test. That definitely should not matter, but you could nevertheless try running all test cases instead.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c13) by @mkorpela on 17 Dec 2012*


Seems that you don't have pybot command in your example.. That is strange.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c14) by @pekkaklarck on 18 Dec 2012*


Which `Execution Profile` you use for running tests? I got this kind of error when using `custom script` without actually defining what script to use. If you have that profile in use, try changing it to `pybot`. We should also provide a better error in this situation.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c15) by martijn.... on 19 Dec 2012*


I react first on #12. Yes, my simple test fail after a fresh installation. The argument file is indeed with newlines.

I created the simple test case because my original tests did not work. I could thus reproduce my problem without using our proprietary Python code. I then started ride from a directory that contains only the simple test case. There was one test suite and it contained the simple test. I therefore cannot follow-up on your last suggestion: there are no other tests to include.

I wonder how fresh my new installation is. I may start a new virtual box by copying my original one with ride 0.46 or ride 0.47. Then I can de-install robot framework and ride and install again. This way I can check I did nothing wrong. Do you think this makes sense?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c16) by martijn.... on 19 Dec 2012*


Reaction to #13. Yes, that is true. The word "pybot" is missing in the ride console.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c17) by martijn.... on 19 Dec 2012*


Reaction on #14. I set the execution profile to pybot and then my test worked. I will try now on my original tests.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c18) by martijn.... on 20 Dec 2012*


I have tried my original tests again. The original problem persists. The tests start running, but after some time the console becomes gray. The console is also unresponsive. This happened before running the selected tests was finished, and in the midst of a test case. While the screen is gay, no tests are running anymore.

I am still interested in a solution for this issue, but I must inform you that I will be on holiday next week and the week thereafter. I will be back on Monday January 7.

With kind regards,

Martijn Dirkse

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c19) by @mkorpela on 31 Dec 2012*


Hi Martijn,

You could try to execute ride with following command line option (this will allow you to interact with a debug console during ride run).

ride.py --debugconsole

once the problem is occurring type:
&gt;&gt;&gt; print_stacks()

this will show what all the threads in RIDE process are currently doing.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c20) by martijn.... on 7 Jan 2013*


Dear Mikko and Pekka,

I am sorry, but I have no more time to work on this issue. Other issues within our development team have a higher priority and therefore I will not follow-up on your last remark. I hope you find another way to analyze this issue.

With kind regards,

Martijn Dirkse

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c21) by @pekkaklarck on 9 Jan 2013*


Closing because we don't have any change to debug this further. If this was a memory issue, upgrading to RF 2.7.6 might help because it contains a fix for a pretty bad memory leak.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c22) by yogitach... on 13 Oct 2014*


Were you able to fix it? Even I am facing same issue

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1145#c23) by martijn.... on 13 Oct 2014*


Dear Robot Framework developer,

Since I reported this issue, I updated my Ubuntu installation. By then I used Ubuntu 10, now I use Ubuntu 12. I do not know whether I still have this issue, because I have not worked with Robot Framework for some time. I will put this email on my todo list. I hope I can find the time to test it under ubuntu 12.

With kind regards,

Martijn Dirkse
